lib/awful/menu.lua

```diff
@@ -239,17 +241,37 @@ local function check_access_key(self, key)
    end
 end
 
+function menu:select_next(direction, current)
+    local count = #self.items
+    if count < 1 then
+        return
+    end
+
+    local index = current or self.sel or 0
+    for _ = 1, count do
+        index = index + (direction or 1)
+        if index < 1 then
+            index = count
+        elseif index > count then
+            index = 1
+        end
```

Comment:
I think we are trying to loop over the items table as in a circular array.

Please remove this for loop and use the modulo operator instead to get the new index.

Reply:
The loop is necessary, because we want to cycle until we find the first enabled item. I use `for` to loop `count` times instead of `while true` to make sure we don't loop forever if all items are disabled.
Using just a modulo would not solve this.